### ISSUE TITLE
Elvish shell on Windows receives pasted text slowly (ConHost/ConPTY related?)

### ISSUE THREAD

**Issue by lilacse** [microsoft/terminal#16549]
### Windows Terminal version

1.18.3181.0

### Windows build number

Microsoft Windows [Version 10.0.22621.2861]

### Other Software

Elvish (0.20.0-dev.0.20240108164435-9112eb1ab2f4+official) (latest HEAD)

### Steps to reproduce

Run Elvish on Windows, and paste a sufficiently long chunk of text into the shell's prompt, about 1k to 2k characters. 

### Expected Behavior

The entire pasted content to be received by the shell in one go, matching the behaviour exhibited by the shell on Unix-like environments and WSL2. 

### Actual Behavior

The pasted content is received by the shell gradually. 

https://github.com/microsoft/terminal/assets/58763824/f3bc2323-2c99-4ba6-ace5-76bf6fde7d02

(This happens on both Windows Terminal and Alacritty)

I previously reported this to the Elvish repo (https://github.com/elves/elvish/issues/1744), and the discussion pointed that this might be an issue with Windows's ConHost/ConPTY and bracketed pasting. Hopefully this is the right place for the issue :)

**Comment by lhecker** [comment]
Yep this is the right place! It's not an issue with our code however and so I'll close this issue. It's also not _really_ an issue with elvish either and fixing it requires just 1 line change in elvish. I'll comment in the linked elvish issue how to fix this. 😊
